### PR TITLE
Preparatory refactoring for adding SC Watcher

### PR DESCRIPTION
**Pull request by andreibancioiu**
Preparatory refactoring for adding SC Watcher:

 - Removed deprecated `serviceContainer`
 - Remove `dbIndexer` global variable
 - Adjusted injection of `indexer` and `tpsBenchmark`

### CODE REVIEW

File: cmd/node/main.go [hunk 1]
```diff
@@ -1250,9 +1234,7 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		return fmt.Errorf("%w when adding nodeShufflerOut in hardForkTrigger", err)
 	}
 
-	var elasticIndexer indexer.Indexer
-	if !check.IfNil(coreServiceContainer) && !check.IfNil(coreServiceContainer.Indexer()) {
-		elasticIndexer = coreServiceContainer.Indexer()
+	if !elasticIndexer.IsNilIndexer() {
 		elasticIndexer.SetTxLogsProcessor(processComponents.TxLogsProcessor)
```
Comment (sasurobert) [inline cmd/node/main.go:1238]: we need to rethink the Indexer interface - have one common - which gets all the data -> after that we could RegisterHandlers. One of those can be elastic search.

File: integrationTests/testProcessorNode.go [hunk 2]
```diff
@@ -15,6 +15,8 @@ import (
 	"github.com/ElrondNetwork/elrond-go/core"
 	"github.com/ElrondNetwork/elrond-go/core/accumulator"
 	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/indexer"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline integrationTests/testProcessorNode.go:19]: do not import mock from another package.

File: integrationTests/testSyncNode.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 
 	"github.com/ElrondNetwork/elrond-go/consensus/spos/sposFactory"
 	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/indexer"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline integrationTests/testSyncNode.go:9]: do not import mock from another package

File: process/block/baseProcess_test.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ import (
 	"time"
 
 	"github.com/ElrondNetwork/elrond-go/core"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline process/block/baseProcess_test.go:14]: do not import mock from another package

File: process/block/export_test.go [hunk 5]
```diff
@@ -5,6 +5,7 @@ import (
 
 	"github.com/ElrondNetwork/elrond-go/core"
 	"github.com/ElrondNetwork/elrond-go/core/check"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline process/block/export_test.go:8]: do not import mock from another package

File: process/block/metablock_test.go [hunk 6]
```diff
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	"github.com/ElrondNetwork/elrond-go/core"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline process/block/metablock_test.go:13]: do not import mock from another package

File: process/block/shardblock_test.go [hunk 7]
```diff
@@ -13,7 +13,7 @@ import (
 	"time"
 
 	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/indexer"
+	coreMock "github.com/ElrondNetwork/elrond-go/core/mock"
```
Comment (sasurobert) [inline process/block/shardblock_test.go:16]: do not import mock from another package

File: cmd/node/main.go (hunk 1 above)
Comment (andreibancioiu) [reply cmd/node/main.go:1238]: I will try to do some small, small redesign when implementing the `ScWatcher`. Will see how it works.

File: integrationTests/testProcessorNode.go (hunk 2 above)
Comment (andreibancioiu) [reply integrationTests/testProcessorNode.go]: Fixed by moving it to `testscommon`, where we hold another commonly-used mocks.

File: integrationTests/testSyncNode.go (hunk 3 above)
Comment (andreibancioiu) [reply integrationTests/testSyncNode.go]: As above.

File: process/block/baseProcess_test.go (hunk 4 above)
Comment (andreibancioiu) [reply process/block/baseProcess_test.go]: As above.

File: process/block/export_test.go (hunk 5 above)
Comment (andreibancioiu) [reply process/block/export_test.go]: As above.

File: process/block/metablock_test.go (hunk 6 above)
Comment (andreibancioiu) [reply process/block/metablock_test.go]: As above.

File: process/block/shardblock_test.go (hunk 7 above)
Comment (andreibancioiu) [reply process/block/shardblock_test.go]: As above.

**Pull request merged**